fix: Possible null pointer dereference introduced when adding NSS full non-blocking semantics

Resolves: #684035 (RHEL)
Upstream ITS: #6863

--- openldap-2.4.24.orig/libraries/libldap/tls_m.c
+++ openldap-2.4.24/libraries/libldap/tls_m.c
@@ -2272,7 +2272,7 @@
 	}
 
 	if ( p->firsttag == LBER_SEQUENCE ) {
-		if ( *thebyte ) {
+		if ( thebyte ) {
 			*thebyte = p->firsttag;
 		}
 		return 1;
@@ -2769,7 +2769,7 @@
 	struct tls_data		*p;
  	p = tlsm_get_pvt_tls_data( fd );
 
-	if ( !data ) {
+	if ( p == NULL || data == NULL ) {
 		return PR_FAILURE;
 	}
 
